dvc/repo/add.py

```diff
@@ -173,12 +173,20 @@ def _process_stages(
             )
         else:
             from dvc.fs import get_cloud_fs
-            from dvc.objects.transfer import transfer
+            from dvc.objects import save as osave
+            from dvc.objects.stage import stage as ostage
 
             from_fs = get_cloud_fs(repo, url=target)
-            out.hash_info = transfer(
-                out.odb, from_fs, from_fs.path_info, jobs=kwargs.get("jobs"),
+            obj = ostage(
+                out.odb,
+                from_fs.path_info,
+                from_fs,
+                "md5",
+                upload=True,
+                jobs=kwargs.get("jobs"),
             )
+            osave(out.odb, obj, move=False)
+            out.hash_info = obj.hash_info
```

Comment:
I guess we could create a `out.transfer()` method, so we could reuse it with `odb=repo.cloud.odb` in --to-remote as well. But it could be done later in a followup.